plenum/test/consensus/order_service/test_sim_ordering_node_txn.py

```diff
@@ -0,0 +1,266 @@
+        return node_to_demote.name.split(':')[0], \
+               build_demote_node_req(Greeks[node_index][0],
+                                     get_txn_data_from_ledger(
+                                         node_to_demote._write_manager.database_manager.get_ledger(POOL_LEDGER_ID),
+                                         Greeks[node_index][0]))
+    return wrap
+
+
+# "params" equal to seed
+@pytest.fixture(params=range(100))
+def random(request):
+    return DefaultSimRandom(request.param)
+
+
+@pytest.fixture()
+def sim_pool(random):
+    return setup_pool(random)
+
+
+# Get list of random indexes. Count of this indexes is not greater then
+@pytest.fixture()
+def indexes_to_demote(sim_pool, random):
+    return random.sample(range(1, sim_pool.size), sim_pool.size // 2)
+
+
+@pytest.fixture()
+def pool_with_edge_primaries(sim_pool):
+    """
+    Validators are:
+    [A, B, C, D]. On view == 3 primaries should be [D, A]
+    After adding new node we expect, that view_change will be proposed
+    and new primaries would be [A, B], because primaries will be selected
+    for last stable NodeReg ([A, B, C, D])
+    """
+    expected_view_no = sim_pool.size - 1
+    sim_pool.initiate_view_change(expected_view_no)
+    sim_pool.timer.wait_for(lambda: all([n._data.view_no == expected_view_no for n in sim_pool.nodes]))
+    sim_pool.timer.wait_for(lambda: all([not n._data.waiting_for_new_view for n in sim_pool.nodes]))
+    # Write primaries to audit
+    order_requests(sim_pool)
+    return sim_pool
+
+
+def test_node_txn_add_new_node(node_req_add, sim_pool, random):
+    # Step 1. Prepare NODE requests and some of params to check
+    # Count of NODE requests is random but less then pool size
+    pool_reqs = [node_req_add(sim_pool.size + i) for i in range(random.integer(1, sim_pool.size - 1))]
+    domain_reqs = create_requests(DOMAIN_REQ_COUNT)
+    reqs = pool_reqs + domain_reqs
+    shuffle(reqs)
+    sim_pool.sim_send_requests(reqs)
+
+    current_view_no = sim_pool.view_no
+    current_pool_ledger_size = get_pools_ledger_size(sim_pool, ledger_id=POOL_LEDGER_ID)
+    current_domain_ledger_size = get_pools_ledger_size(sim_pool, ledger_id=DOMAIN_LEDGER_ID)
+    expected_view_no = current_view_no + 1
+    expected_node_reg = sim_pool.validators + [Greeks[sim_pool.size + i][0] for i in range(len(pool_reqs))]
+
+    # Step 2. Start requests ordering
+    random_interval = random.integer(10, 20) * 100
+    RepeatingTimer(sim_pool.timer, random_interval, partial(order_requests, sim_pool))
+
+    # Step 3. Initiate view change process during request's ordering
+    for node in sim_pool.nodes:
+        sim_pool.timer.schedule(random_interval + 1000,
+                                partial(node._view_changer.process_need_view_change, NeedViewChange(view_no=1)))
+
+    # Step 4. Wait for VC completing
+    for node in sim_pool.nodes:
+        sim_pool.timer.wait_for(lambda: node._view_changer._data.view_no == 1)
```

Comment:
Do we expect only 1 view change here?
Should we try to do more than one?
Also adding/demoting of nodes will trigger a view change, so it can do multiple view changes even now, can't it?